pkg/discovery/dns/provider_test.go

```diff
@@ -37,40 +36,67 @@ func TestProvider(t *testing.T) {
 	result := prv.Addresses()
 	sort.Strings(result)
 	testutil.Equals(t, []string(nil), result)
+	testutil.Equals(t, 1, promtestutil.CollectAndCount(prv.resolverAddrs))
+	testutil.Equals(t, float64(0), promtestutil.ToFloat64(prv.resolverAddrs.WithLabelValues("any+x")))
 
 	prv.Resolve(ctx, []string{"any+a", "any+b", "any+c"})
 	result = prv.Addresses()
 	sort.Strings(result)
 	testutil.Equals(t, ips, result)
+	testutil.Equals(t, 3, promtestutil.CollectAndCount(prv.resolverAddrs))
```

Comment:
Could we please add a comment in this test case to clarify the repetition of test cases. I believe, essentially it is for testing determinist resolution.

Reply:
Wait. I cannot see the repetition.. We check same stuff and we cache a lot, so it's definitely not repetition.